File: java/Samples/BiometricStandards/CS/FMRecordSampleCS/SelectedFmrItems.cs
using Neurotec.Biometrics.Standards;

namespace Neurotec.Samples
{
	public class SelectedFmrItem
	{
		protected readonly FmrFingerView FingerView;
		protected readonly int Index;

		public SelectedFmrItem(FmrFingerView fingerView, int index)
		{
			FingerView = fingerView;
			Index = index;
		}
	}

	public class SelectedFmrDelta : SelectedFmrItem
	{
		public SelectedFmrDelta(FmrFingerView fingerView, int index)
			: base(fingerView, index)
		{
		}

		public FmrDelta Delta
		{
			get { return FingerView.Deltas[Index]; }
			set { FingerView.Deltas[Index] = value; }
		}
	}

	public class SelectedFmrCore : SelectedFmrItem
	{
		public SelectedFmrCore(FmrFingerView fingerView, int index)
			: base(fingerView, index)
		{
		}

		public FmrCore Core
		{
			get { return FingerView.Cores[Index]; }
			set { FingerView.Cores[Index] = value; }
		}
	}

	public class SelectedFmrMinutia : SelectedFmrItem
	{
		public SelectedFmrMinutia(FmrFingerView fingerView, int index)
			: base(fingerView, index)
		{
		}

		public FmrMinutia Minutia
		{
			get { return FingerView.Minutiae[Index]; }
			set { FingerView.Minutiae[Index] = value; }
		}
	}
}

File: java/Samples/BiometricStandards/VB/FMRecordSampleVB/SelectedFmrItems.vb
Imports Neurotec.Biometrics.Standards

Public Class SelectedFmrItem
	Protected ReadOnly FingerView As FmrFingerView
	Protected ReadOnly Index As Integer

	Public Sub New(ByVal fingerView As FmrFingerView, ByVal index As Integer)
		Me.FingerView = fingerView
		Me.Index = index
	End Sub
End Class

Public Class SelectedFmrDelta
	Inherits SelectedFmrItem
	Public Sub New(ByVal fingerView As FmrFingerView, ByVal index As Integer)
		MyBase.New(fingerView, index)
	End Sub

	Public Property Delta() As FmrDelta
		Get
			Return FingerView.Deltas(Index)
		End Get
		Set(ByVal value As FmrDelta)
			FingerView.Deltas(Index) = value
		End Set
	End Property
End Class

Public Class SelectedFmrCore
	Inherits SelectedFmrItem
	Public Sub New(ByVal fingerView As FmrFingerView, ByVal index As Integer)
		MyBase.New(fingerView, index)
	End Sub

	Public Property Core() As FmrCore
		Get
			Return FingerView.Cores(Index)
		End Get
		Set(ByVal value As FmrCore)
			FingerView.Cores(Index) = value
		End Set
	End Property
End Class

Public Class SelectedFmrMinutia
	Inherits SelectedFmrItem
	Public Sub New(ByVal fingerView As FmrFingerView, ByVal index As Integer)
		MyBase.New(fingerView, index)
	End Sub

	Public Property Minutia() As FmrMinutia
		Get
			Return FingerView.Minutiae(Index)
		End Get
		Set(ByVal value As FmrMinutia)
			FingerView.Minutiae(Index) = value
		End Set
	End Property
End Class
